Compilation error `non-primitive cast` when returning nothing in key-released callback

Slint 1.3.0

Probably this is caused by returning none in this function
```
export component SelectableTableView {
    focus_item := FocusScope {
        key-released(event) => {
            debug(event);
            if (!self.visible) {
                return;
            }
            accept
        }
    }
}
```
This works in lsp but compilation throws error
```
error[E0605]: non-primitive cast: `()` as `EventResult`
     --> /home/rafal/Projekty/Rust/czkawka/target/debug/build/krokiet-525b5ee05e6ac701/out/main_window.rs:11186:34
      |
11186 | / ...                    ({
11187 | | ...                        }
11188 | | ...                   ) as _ }
      | |____________________________^ an `as` expression can only be used to convert between primitive types or to coerce to a specific trait object
```
Probably this is commit which caused problem - https://github.com/qarmin/czkawka/pull/1102/commits/796233c54f249d848ba81c8ef585582912fcf740

--- a/internal/compiler/passes/resolving.rs
+++ b/internal/compiler/passes/resolving.rs
@@ -208,7 +208,11 @@ impl Expression {
         ctx: &mut LookupCtx,
     ) -> Expression {
         let return_type = ctx.return_type().clone();
-        Expression::ReturnStatement(node.Expression().map(|n| {
+        let e = node.Expression();
+        if e.is_none() && !matches!(return_type, Type::Void | Type::Invalid) {
+            ctx.diag.push_error(format!("Must return a value of type '{return_type}'"), &node);
+        }
+        Expression::ReturnStatement(e.map(|n| {
             Box::new(Self::from_expression_node(n, ctx).maybe_convert_to(
                 return_type,
                 &node,

--- a/internal/compiler/tests/syntax/basic/return.slint
+++ b/internal/compiler/tests/syntax/basic/return.slint
@@ -13,4 +13,22 @@ export X := Rectangle {
         return 42phx;
 //      ^error{Cannot convert physical-length to string. Divide by 1phx to convert to a plain number}
     }
+
+
+    // Issue 3962
+    FocusScope {
+        key-released(event) => {
+            if (!self.visible) {
+                return;
+//              ^error{Must return a value of type 'enum EventResult'}
+            }
+            accept
+        }
+
+        enabled: {
+            return;
+//          ^error{Must return a value of type 'bool'}
+            true;
+        }
+    }
 }

